go/adbc/driver/bigquery/connection.go

```diff
@@ -617,7 +617,32 @@ func (c *connectionImpl) getTableSchemaWithFilter(ctx context.Context, catalog *
 		metadata["MaterializedView.Query"] = md.MaterializedView.Query
 		metadata["MaterializedView.RefreshInterval"] = md.MaterializedView.RefreshInterval.String()
 		metadata["MaterializedView.AllowNonIncrementalDefinition"] = strconv.FormatBool(md.MaterializedView.AllowNonIncrementalDefinition)
-		metadata["MaterializedView.MaxStaleness"] = md.MaxStaleness.String()
+		if md.MaxStaleness != nil {
+			metadata["MaterializedView.MaxStaleness"] = md.MaxStaleness.String()
+		}
+	}
+	if md.TimePartitioning != nil {
+		// "DAY", "HOUR", "MONTH", "YEAR"
+		metadata["TimePartitioning.Type"] = string(md.TimePartitioning.Type)
```

Comment:
do type and field need to come in pairs? if yes, should we check they are both set or not

Reply:
`Type` is non-nullable (i.e. it has a default)